### PR TITLE
planner: recover table/schema only needs create and drop privilege

### LINKED ISSUE #40633: Fine grained privilege to RECOVER TABLE
## Feature Request

**Is your feature request related to a problem? Please describe:**
A user has CREATE and DROP privilege for tables in a particular schema (ON schema.*). He accidentally dropped a table. He wanted to RECOVER the table, but he didn't have SUPER privilege. Before he was able to contact DBA with SUPER privilege, the GC life time was over and DBA couldn't RECOVER the table either.

**Describe the feature you'd like:**
If a user has CREATE and DROP privilege for a particular table, (s)he could also be allowed to RECOVER that table. Or perhaps there could be a RECOVER privilege that would be bound to tables (like CREATE and DROP) so that it could be granted to users like CREATE and DROP privileges.

**Describe alternatives you've considered:**
Increase GC life time to 4 days or more which is max. expected time before DBA can be contacted and still have enough time to act. This could affect query performance though.

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #40633 

Problem Summary: Follow spec, refine privilege for `flashback table/schema` ddl commands.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by dveeden — APPROVED** [review]
Looks good. Note that backporting to versions before v6.1.0 might not be safe because of `lower-case-table-names`.

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -4727,9 +4727,35 @@ func (b *PlanBuilder) buildDDL(ctx context.Context, node ast.DDLNode) (Plan, err
...
+	case *ast.FlashBackTableStmt:
+		if b.ctx.GetSessionVars().User != nil {
+			authErr = ErrTableaccessDenied.GenWithStackByArgs("CREATE", b.ctx.GetSessionVars().User.AuthUsername,
+				b.ctx.GetSessionVars().User.AuthHostname, v.Table.Name.L)
+		}
+		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.CreatePriv, v.Table.Schema.L, v.Table.Name.L, "", authErr)
+		if b.ctx.GetSessionVars().User != nil {
+			authErr = ErrTableaccessDenied.GenWithStackByArgs("DROP", b.ctx.GetSessionVars().User.AuthUsername,
+				b.ctx.GetSessionVars().User.AuthHostname, v.Table.Name.L)
+		}
```
Comment (xhebox) [inline planner/core/planbuilder.go:4750]: Why check twice?

Comment (Defined2014) [reply planner/core/planbuilder.go:4750]: this one for `DROP`, another one for `Create`. User should have both two privilege to do `flashback`.

Comment (Defined2014) [reply planner/core/planbuilder.go:4750]: And the stmt `RecoverTableStmt` and `FlashBackTableStmt` are not same. So we have two cases here.

**Review by xhebox — APPROVED** [review]

**Comment by Defined2014** [comment]
/hold

merge it after v6.6.0

**Pull request merged**